Dashboard/app/js/lib/views/data/question-answer-view.js

```diff
@@ -1,138 +1,141 @@
-    this.set('value', this.content.get('value'));
+  isVideoType: function(){
+    return this.get('questionType') === 'VIDEO';
+  }.property('this.questionType'),
 
-    if (this.get('isNumberType')) {
-      this.set('numberValue', this.content.get('value'));
+  optionsList: function(){
+    var c = this.content;
+    if (Ember.none(c)) {
+      return [];
     }
 
-    if (this.get('isDateType') && !Ember.none(this.content.get('value'))) {
-      date = new Date(parseInt(this.content.get('value'), 10));
-      this.set('date', formatDate(date));
+    var questionId = c.get('questionID');
+
+    var options = FLOW.store.filter(FLOW.QuestionOption, function (item) {
+      return item.get('questionId') === +questionId;
+    });
+
+    optionArray = options.toArray();
+    optionArray.sort(function (a, b) {
+        return a.get('order') - b.get('order');
+    });
+
+    tempList = [];
+    optionArray.forEach(function (item) {
+      tempList.push(item.get('text'));
+    });
+    return tempList;
+  }.property('this.content'),
+
+  content: null,
+
+  inEditMode: false,
+
+  isNotEditable: function(){
+    var type = this.get('questionType');
+    return (type == 'GEO' || type == 'PHOTO' || type == 'VIDEO' || type == 'GEOSHAPE');
+  }.property('this.questionType'),
+
+  date: null,
+
+  numberValue: null,
+
+  photoUrl: function(){
+    var c = this.content;
+    if (!Ember.empty(c.get('value'))) {
+      return FLOW.Env.photo_url_root + c.get('value').split('/').pop();
     }
-    if (this.get('isOptionType') && !Ember.none(this.content.get('value'))) {
-      options = FLOW.store.filter(FLOW.QuestionOption, function (item) {
-        return item.get('questionId') == questionId;
-      });
-
-      optionArray = options.toArray();
-      optionArray.sort(function (a, b) {
-    	  return a.get('order') - b.get('order');
-      });
-
-      tempList = [];
-      optionArray.forEach(function (item) {
-        tempList.push(Ember.Object.create({
-          isSelected: false,
-          value: item.get('text')
-        }));
-      });
-      this.set('optionsList', tempList);
-
-      // set answer
-      qaValue = this.content.get('value');
-      this.get('optionsList').forEach(function (item) {
-        if (item.get('value') == qaValue) {
-          choice = item;
-        }
-      });
-      this.set('optionChoice', choice);
+  }.property('this.content,this.isPhotoType,this.isVideoType'),
+
+  questionType: function(){
+    if(this.get('question')){
+      return this.get('question').get('type');
     }
-    if ((this.get('isPhotoType') || this.get('isVideoType')) && !Ember.empty(this.content.get('value'))) {
-      // Since photos have a leading path from devices that we need to trim
-      this.set('photoUrl', FLOW.Env.photo_url_root + this.content.get('value').split('/').pop());
+  }.property('this.question'),
+
+  question: function(){
+    var c = this.get('content');
+    if (c) {
+      var questionId = this.content.get('questionID');
```

Comment:
Given my little knowledge on Ember, this is more of a naive question, but isn't `this.content` pointing to the same variable as the previously declared `c`?